test/test_torch.py

```diff
@@ -16368,53 +16388,45 @@ def _test(use_out, row_major, incx, incy, lda_tail):
             y_storage = torch.full((o, incy), float('nan'), device=device, dtype=dtype)
             y = y_storage[:, 0].copy_(y_data)
 
-            if use_out:
-                out = torch.addmv(y, a, x)
-            else:
-                out = torch.empty_like(y)
-                torch.addmv(y, a, x, out=out)
-
-            self.assertEqual(out, control, atol=1.e-4, rtol=0)
-
-        for use_out, row_major, incx, incy, lda_tail in product((False, True), (False, True), (1, 2), (1, 2), (0, 1)):
-            _test(use_out, row_major, incx, incy, lda_tail)
+            self._test_addmm_addmv(torch.addmv, y, a, x)
 
-    def _test_addmm(self, M, m1, m2):
-        dtype = M.dtype
-        numpy_dtype = dtype
-        if dtype in {torch.bfloat16}:
-            numpy_dtype = torch.float
-        if dtype.is_complex:
-            alpha = 0.9 + 0.3j
-            beta = 0.5 + 0.6j
-        else:
-            alpha = 1.2
-            beta = 0.8
-        res1 = torch.addmm(M, m1, m2, alpha=alpha, beta=beta)
-        res2 = torch.full_like(res1, math.nan)
-        torch.addmm(M, m1, m2, alpha=alpha, beta=beta, out=res2)
-        res3 = (beta * M).to(numpy_dtype).cpu().numpy() + alpha * (
-            m1.to(numpy_dtype).cpu().numpy() @ m2.to(numpy_dtype).cpu().numpy())
-        res3 = torch.from_numpy(res3).to(dtype)
-        self.assertEqual(res1, res2)
-        self.assertEqual(res1, res3)
+        for row_major, incx, incy, lda_tail in product((False, True), (1, 2), (1, 2), (0, 1)):
+            _test(row_major, incx, incy, lda_tail)
 
     @precisionOverride({torch.double: 1e-8, torch.float: 1e-4, torch.bfloat16: 0.6,
                         torch.half: 1e-1, torch.cfloat: 1e-4, torch.cdouble: 1e-8})
     @dtypesIfCUDA(*torch.testing.get_all_complex_dtypes(), *torch.testing.get_all_fp_dtypes(include_bfloat16=False))
     @dtypes(*torch.testing.get_all_complex_dtypes(), *torch.testing.get_all_fp_dtypes())
     @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
     def test_addmm(self, device, dtype):
-        M = torch.randn(10, 25).to(device=device, dtype=dtype)
-        m1 = torch.randn(10, 50).to(device=device, dtype=dtype)
-        m2 = torch.randn(50, 25).to(device=device, dtype=dtype)
-        self._test_addmm(M, m1, m2)
+        M = torch.randn(10, 25, device=device).to(dtype)
```

Comment:
So `(..., device=device, dtype=dtype)` does not work for bfloat16 on cpu? Interesting. `rand` works btw.